Adding compatibility for OctoberCMS v3

sanPuerquitoProgramador opened the pull request:
:) Thanks for this great tool

sanPuerquitoProgramador committed:
Adding compatibility for OctoberCMS v3

sanPuerquitoProgramador committed:
modificacion para tomar solo las columnas que se necesitan

sanPuerquitoProgramador committed:
Validacion y conversión de fechas

sanPuerquitoProgramador committed:
Updated README

sanPuerquitoProgramador committed:
Validando fechas vacias. Si una celda no tiene valor se ignora

rubenvanerk commented:
Hey @sanPuerquitoProgramador, I'd still be interested to merge this in if I get the time. Would you be interested to open a new PR to support V3?

sanPuerquitoProgramador committed:
- Limpieza de codigo con llamadas use
 - Funcionalidad para usar solo las columnas necesarias del excel a importar

sanPuerquitoProgramador committed:
sustitución de metodos deprecados

sanPuerquitoProgramador committed:
- Columns to Keep ahora es opcional

sanPuerquitoProgramador committed:
- Se evalua si la celda está vacia o no para establecer el valor en nulo
 - Se evalua si la celda contiene una fecha y se da formato estándar para la integración a la BD de manera natural sin esperar que el controlador de importación lo evalue

sanPuerquitoProgramador commented:
Here is the translated message:

---

Hi there, sorry for deleting the previous PR, but I realized that what I wanted to submit was not ready yet.

I have added compatibility for OctoberCMS v3 and included several extra features.

- Now it checks if the cell contains a date, and if so, it formats it to the standard format for database insertion. For example, if the date comes in Spanish format without time (d-m-Y), it is automatically formatted to (Y-m-d H:i:s).

- I realized that there are occasions when you might have N number of columns in the Excel file and only want to import the information from columns A, B, C, H, J, L.

For those cases, I have added a variable `$columnsToKeep ` that accepts three states that are set from the controller's constructor of the plugin.

```php
/**
 * __construct the controller
 */
public function __construct()
{
    parent::__construct();

    // Define the columns to keep
    $columnsToKeep = ['A', 'B', 'C', 'H', 'J', 'L'];

    $this->asExtension('ExcelImportExportController')->setColumnsToKeep($columnsToKeep);

    BackendMenu::setContext('Author.Plugin', 'Library', 'books');
}
```

Another option is to set the value to "all".

```php
/**
 * __construct the controller
 */
public function __construct()
{
    parent::__construct();

    // Define the columns to keep 
    $columnsToKeep = "all";

    $this->asExtension('ExcelImportExportController')->setColumnsToKeep($columnsToKeep);

    BackendMenu::setContext('Author.Plugin', 'Library', 'books');
}
```

Or the simplest one is not to set the variable.

```php
/**
 * __construct the controller
 */
public function __construct()
{
    parent::__construct();

    BackendMenu::setContext('Author.Plugin', 'Library', 'books');
}
```

I hope you find these features useful.

Have a great day and thank you very much for your tool. It helped me a lot with my project.

rubenvanerk commented:
Thanks! Can you please leave out the parts where you mention this is a fork?
Also let's focus on v3 compatibility for this PR, feel free to suggest new features in separate PR's
